src/app/hooks/useGlean.ts

```diff
@@ -62,6 +67,11 @@ export const useGlean = (experimentationId?: string) => {
     // Instead of the specific type definitions we generated in the npm script
     // `build-glean-types`, Glean takes a non-specific "ExtraArgs" type as
     // parameter to `record`.
+
+    // Record the `plan_tier` key on all events.
+    // @ts-expect-error Not present in map, but exists on all events.
+    data.plan_tier = isPremiumUser ? "Plus" : "Free";
```

Comment:
I think it's just too much work for TypeScript to infer that `plan_tier` is present on every entry - it would have to create a type intersection for every event (e.g. `click`) event module (e.g. `button`), and _then_ create a type section for each of those.

It's probably better not to expect an error (because it's really not an error, and if they ever find a way to do the inference performantly, it would cause an error), but to just take one of the event module's type definitions and tell TS that that applies regardless of which event module's actually being set:

```suggestion
    // `plan_tier` is set on every metric, but it's too much work for TypeScript
    // to infer that — hence the type assertion.
    (data as GleanMetricMap["button"]["click"]).plan_tier = isPremiumUser
      ? "Plus"
      : "Free";
```

It won't warn us if we ever add an event without adding `plan_tier`, but :shrug: